docs/content/src/learn/tutorials/messaging-integrations/messaging-with-kafka/README.md

```diff
@@ -0,0 +1,272 @@
+        };
+
+        // Serialize the message
+        byte[] serializedMessage = productInfo.toJsonString().toBytes();
+
+        if (productType.toString() == "Fruit") {
+            io:println("ProductManagementService : Sending message to Partition 0");
+            var sendResult = kafkaProducer->send(serializedMessage, "product-price", partition = 0);
+        } else if (productType.toString() == "Vegetable") {
+            io:println("ProductManagementService : Sending message to Partition 1");
+            var sendResult = kafkaProducer->send(serializedMessage, "product-price", partition = 1);
+        }
+
+        response.setJsonPayload({ "Status" : "Success" });
+        var responseResult = caller->respond(response);
+    }
+}
+```
+
+Let us consider the consumer services, `consumer1` and `consumer2`
+
+```ballerina
+// Kafka Consumer Configuration
+kafka:ConsumerConfig consumer1 = {
+    bootstrapServers: "localhost:9092",
+    groupId: "consumer",
+    topics: ["product-price"],
+    pollingIntervalInMillis: 1000,
+    partitionAssignmentStrategy: "org.apache.kafka.clients.consumer.RoundRobinAssignor"
+
+    listener kafka:Consumer productConsumer1 = new (consumer1);
+};
+```
+
+We make use of a partitionAssignmentStrategy to assign the consumer to a particular topic partition.
+
+Let us now see the complete implementation of a consumer.
+
+#### consumer1.bal
+```ballerina
+import ballerina/kafka;
+import ballerina/io;
+import ballerina/lang.'string as strings;
+
+// Kafka Consumer Configuration
+kafka:ConsumerConfig consumer1 = {
+    bootstrapServers: "localhost:9092",
+    groupId: "consumer",
+    topics: ["product-price"],
+    pollingIntervalInMillis: 1000,
+    partitionAssignmentStrategy: "org.apache.kafka.clients.consumer.RoundRobinAssignor"
+};
+
+// Kafka Listener
+listener kafka:Consumer productConsumer1 = new (consumer1);
+
+// Service that listens to the particular topic
+service productConsumerService1 on productConsumer1 {
+    // Trigger whenever a message is added to the subscribed topic
+    resource function onMessage(kafka:Consumer productConsumer, kafka:ConsumerRecord[] records) returns error? {
+        foreach var entry in records {
+            byte[] serializedMessage = entry.value;
+            string|error stringMessage = strings:fromBytes(serializedMessage);
+
+            if (stringMessage is string) {
+                io:StringReader sr = new (stringMessage);
+                json jsonMessage = check sr.readJson();
+
+                io:println("ProductConsumerService1 : Product Received");
+                io:println("Name : ", jsonMessage.Name);
+                io:println("Price : ", jsonMessage.Price);
+            }
+        }
+    }
+}
+```
+
+In the above code, resource function `onMessage` will be triggered whenever a message published to the topic specified.
+
+
+## Deployment
+
+Once you are done with the development, you can deploy the services using any of the methods listed below.
```

Comment:
You have only listed one.

Reply:
I have made the change to one method for now, but I suspect there will be more methods that will be added to this section in the future